eZ/Publish/SPI/Search/Content/IndexerGateway.php

```diff
@@ -40,6 +40,18 @@ public function getContentInSubtree(string $locationPath, int $iterationCount):
      */
     public function countContentInSubtree(string $locationPath): int;
 
+    /**
+     * @throws \Doctrine\DBAL\Exception
```

Comment:
As far as the indexing is concerned, I'm not sure if we want to declare that this can throw this exception. The fact that we are using Doctrine is irrelevant for indexing (at least for the interface). @alongosz ?